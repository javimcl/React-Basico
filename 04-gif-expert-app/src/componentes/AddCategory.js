import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {

        setInputValue(e.target.value);
      //  console.log('llamado handle')

    }

    const handleSubmit = (e) => {
        e.preventDefault();
     //   console.log('submit')
        //const {setCategories} = props; 

        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }

        
       // console.log(setCategories);

    }

    return (

        <form onSubmit={ handleSubmit }>
            {/* <h1>{inputValue}</h1> */}
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}

            />
        </form>

    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}