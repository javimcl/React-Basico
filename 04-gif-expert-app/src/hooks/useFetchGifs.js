import { useState } from "react";
import { useEffect } from "react/cjs/react.development";
import { getGifs } from "../helpers/getGifs";



export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        getGifs(category)
            .then(imgs => {
               // setTimeout(() => {
                    setState({
                        data: imgs,
                        loading: false
                    });
               // }, 3000);
            })

    }, [category])

    // setTimeout(() => {
    //     setState({
    //         data: [1,2,3,4,5],
    //         loading:false
    //     })
    // }, 3000);

    return state; //{data:[], loading:true}
}
