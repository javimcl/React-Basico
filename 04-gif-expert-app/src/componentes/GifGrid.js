import React from 'react'
//import React, { useState, useEffect } from 'react'
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    //    const [count, setCount] = useState(0);

    const { data: images, loading } = useFetchGifs(category);
    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     // getGifs();
    //     getGifs(category)
    //         .then(setImages);
    //     // .then(imgs => setImages(imgs));
    // }, [category]);


    // const getGifs = async () => {
    //     const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category)}&limit=10&api_key=yQDsU2RIFRTYsbt5aZ37IG4b2hQftQlN`
    //     const resp = await fetch(url);
    //     const { data } = await resp.json();

    //     const gifs = data.map(img => {
    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url
    //         }
    //     })

    //     console.log(gifs);
    //     setImages(gifs);

    // }

    //getGifs();

    return (
        <>
            <h3 className='animate__animated animate__swing'>{category}</h3>
            { loading && <p className='animate__animated animate__heartBeat'>Loading</p>}
            <div className='carGrid'>

                {/* <ol> */}
                {
                    images.map(img =>
                        <GifGridItem key={img.id} {...img} />
                        // <li key={id}>{title}</li>
                    )
                }
                {/* </ol> */}

                {/* <h3>{ count }</h3>
            <button onClick={() => setCount(count + 1)}></button> */}
            </div>
        </>
    )
}
