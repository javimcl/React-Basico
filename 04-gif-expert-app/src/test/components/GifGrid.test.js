import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import { GifGrid } from '../../componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba de <GifGrid/>', () => {

    const category = 'One Punch';
    

    test('debe mostrar el componente correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFecthGifs', () => {
        
        const gifs = [{
            id:'ABC',
            url:'htpp:',
            title:'Cualquiera'
        },
    
        {
            id:'DFG',
            url:'htpp:',
            title:'Cualquiera'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/> );

       expect(wrapper).toMatchSnapshot();       
       expect(wrapper.find('p').exists()).toBe(false);
       expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
