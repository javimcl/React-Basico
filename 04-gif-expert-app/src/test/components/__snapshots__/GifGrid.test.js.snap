// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifGrid/> debe de mostrar items cuando se cargan imagenes useFecthGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing"
  >
    One Punch
  </h3>
  <div
    className="carGrid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquiera"
      url="htpp:"
    />
    <GifGridItem
      id="DFG"
      key="DFG"
      title="Cualquiera"
      url="htpp:"
    />
  </div>
</Fragment>
`;

exports[`Prueba de <GifGrid/> debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="carGrid"
  />
</Fragment>
`;
